autoload/lsp/internal/work_done_progress.vim

```diff
@@ -0,0 +1,45 @@
+let s:lsp_progress = {
+      \ 'server': '',
+      \ 'token': '',
+      \ 'title': '',
+      \ 'messages': '',
+      \ 'percentage': 100,
+      \ }
+
+function! lsp#internal#work_done_progress#_enable() abort
+    if !g:lsp_work_done_progress_enabled | return | endif
+
+    let s:Dispose = lsp#callbag#pipe(
+          \ lsp#stream(),
+          \ lsp#callbag#filter({x->has_key(x, 'response') && has_key(x['response'], 'method')
+          \  && x['response']['method'] ==# '$/progress' && has_key(x['response'], 'params')
+          \  && has_key(x['response']['params'], 'value') && type(x['response']['params']['value']) == type({})}),
+          \  lsp#callbag#subscribe({'next': function('s:handle_work_done_progress')})
```

Comment:
Can we change to this so it aligns with https://github.com/prabirshrestha/vim-lsp/pull/978/files

```vim
 \  lsp#callbag#subscribe({'next': {x->s:handle_work_done_progress(x['server'], x['response']}})
```

This allows the code to be self documented.

Reply:
Thank you for comment.
I fixed it.